lesson3.md

```diff
@@ -0,0 +1,837 @@
+<!-- .slide: id="lesson3" -->
+
+# Basic Frontend - Spring 2020
+
+Lesson 3, Tuesday, 2020-09-01
+
+
+---
+
+
+### what we learned last time
+
+* Variables are pointers to values
+
+* We can use operators with variables just like with values
+
+* Use `<script src="xxx.js"></script>` to add a JavaScript file to your HTML webpage.
+
+* Use `//` to write comments.
+
+* Use `console.log(...)` to output values to the console.
+
+
+---
+
+
+### We had a HomeWork
+
+You have an online shop, which sells hats and t-shirts. One of the customers wants to buy 2 hats and 3 t-shirts.
+
+A hat costs 3.99€ and a t-shirt costs 9.99€. What is the total cost?
+
+Create all the variables needed, and output the final result to the console.
+
+
+---
+
+
+### Homework solution
+
+```js
+let hatPrice = 3.99;
+let tShirtPrice = 9.99;
+
+let hatAmount = 2;
+let tShirtAmount = 3;
+
+let totalHatPrice = hatAmount * hatPrice;
+let totalTShirtPrice = tShirtAmount * tShirtPrice;
+
+let totalCost = totalHatPrice + totalTShirtPrice;
+
+console.log(totalCost);
+```
+
+
+---
+
+
+### Recap,  What does it print?
+
+```js
+let name = "Sevtap";
+console.log("Hello " + name);
+```
+
+Answer: It prints "Hello Sevtap" <!-- .element: class="fragment" -->
+
+
+---
+
+
+### What does it print?
+
+```js
+let greeting = "Hello";
+let name = "Sevtap";
+console.log(greeting + " " + firstName);
+```
+
+Answer: Error, "firstName" is not a variable <!-- .element: class="fragment" -->
+
+
+---
+
+
+### What does it print?
+
+```js
+console.log("It is " + temperature + " degrees");
+let temperature = 22;
+```
+
+Answer: Error, "temperature" is not available <!-- .element: class="fragment" -->
+
+
+---
+
+
+What do you think `x` and `y` are at the end of the program?
+
+```js
+let x = 42;
+let y = x;
+x = 43;
+console.log(x);
+console.log(y);
+```
+
+Answer: x is 43, y is 42 <!-- .element: class="fragment" -->
+
+So, Why?  <!-- .element: class="fragment" -->
+
+
+---
+
+
+### Bass by value
+
+* This is called *pass by value*.
+* All basic data types (number, string, ...) are passed by value.
+* If you assign a variable to another variable, its value is __copied__.
+
```

Comment:
Maybe show another example?

```suggestion
```js
let name = "Abdullah";
let name2 = name;
name = "Carlo"; // name is now "Carlo", name2 is still "Abdullah"
```js
```